code/game/machinery/slotmachine.dm

```diff
@@ -12,90 +12,94 @@
 	var/resultlvl = null
 
 /obj/machinery/slot_machine/attack_hand(mob/user as mob)
+	tgui_interact(user)
+
+/obj/machinery/slot_machine/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "SlotMachine", name, 350, 200, master_ui, state)
+		ui.open()
+		ui.set_autoupdate(TRUE)
+
+/obj/machinery/slot_machine/tgui_data(mob/user)
```

Comment:
`ui_data` is replaced with `tgui_data` for separation. In here is where the data for the UI is collected from DM and sent to the JS handler. I moved the code around here to make ID grabbing more responsive, but in most cases this isnt needed.